### PR TITLE
The Events Calendar REST API plugin.

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/81691

This PR adds the TEC REST API plugin in a way similar to the EA Server Mocker.j

### CODE REVIEW

File: tests/_support/ea-server-mocker/src/Options_Page.php [hunk 1]
```diff
@@ -37,39 +37,42 @@ public function register_settings() {
 
 	public function render() {
 		?>
-		<div class="wrap">
-			<h1>Event Aggregator Server Mocker Settings</h1>
+        <div class="wrap">
+            <h1>Event Aggregator Server Mocker Settings</h1>
```
Comment (borkweb) [inline tests/_support/ea-server-mocker/src/Options_Page.php]: This whole chunk looks like it was un-tabbed by one level accidentally. Tab it back?

File: tests/_support/rest-api-tester/assets/js/src/trap-script.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
+(function ($, undefined) {
```
Comment (borkweb) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.js]: Our JS standards use the same spacing as our PHP standards. Add spaces around parens as per those standards :)

File: tests/_support/rest-api-tester/assets/js/src/trap-script.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
+(
```
Comment (borkweb) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.ts]: Ah, the JS above was generated due to TypeScript compilation, perhaps? Side note: we haven't standardized on TypeScript amongst the team so I hesitate at its usage here. That being said, I'm pretty open to a discussion around it. We've talked about adding ES6/Babel to our world - perhaps we do TypeScript instead? I know @barryhughes has been pumped about TypeScript for a while.

File: tests/_support/rest-api-tester/assets/js/src/trap-script.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+	function ( $, undefined ) {
+		var renderjson = require( 'renderjson' );
+
+		var setRequestResponse = function ( data, status, response ) {
```
Comment (borkweb) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.ts]: Remove the space between `function` and `(`

File: tests/_support/rest-api-tester/assets/js/src/trap-script.ts [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+			if ( ! json ) {
+				$response.empty();
+				return;
+			}
+
+			document.getElementById( 'trap-response' ).appendChild( renderjson( JSON.parse( json ) ) );
+		};
+
+		var formatDocumentationJson = function(){
```
Comment (borkweb) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.ts]: Add a space after `)` and before `{`?

File: tests/_support/rest-api-tester/assets/js/src/trap-script.ts (hunk 3 above)
Comment (barryhughes) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.ts]: :point_up: true, it's the _type_ of thing that makes me happy

File: tests/_support/ea-server-mocker/src/Options_Page.php (hunk 1 above)
Comment (lucatume) [inline tests/_support/ea-server-mocker/src/Options_Page.php]: this looks fine in the editor... mmm.

File: tests/_support/rest-api-tester/assets/js/src/trap-script.ts (hunk 3 above)
Comment (lucatume) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.ts]: I've used Typescript here just for convenience (I *really* like strong typehinting) but it is just a source tool. PHP storm supports it out of the box and it's very similar to PHP as syntax.

File: tests/_support/rest-api-tester/assets/js/src/trap-script.js (hunk 2 above)
Comment (lucatume) [inline tests/_support/rest-api-tester/assets/js/src/trap-script.js]: I just found out one can set PHPStorm to use WordPress standards for PHP and then Code Style > Javascript (or Typescript) > Set from... > Language > PHP et voila!

**Comment by lucatume** [comment]
I've fixed the stuff and this should be good for another look.

**Pull request merged**